### PR TITLE
MLR Kickoff

### PR DESCRIPTION
**Pull request by karla-vm**
## Summary
MLR Kickoff PR!

### Description

This PR was co-authored with @oweingrod-coforma 🥇 

We're setting up initial MLR report pages, including: 

- State User Dashboard
- Admin User Dashboard
- MLR Report Page 1
- MLR Report Page 2 (Empty State)
- MLR Report Page 2 (Modal Overlay Form)
- Review and Submit Page

Additionally, the `/mlr` report routes are behind a LaunchDarkly feature flag `mlrReport` to avoid users navigating directly to the report. 

Lastly, this PR also adds some more strict typings for `MCPARReportMetadata` and `MLRReportMetadata` on the backend, with typings for fieldData and formTemplate objects coming soon.

### Related tickets

- MDCT-2212
- MDCT-2230
- MDCT-2234
- MDCT-2226
- MDCT-2223
- MDCT-2233
- MDCT-2225
- MDCT-2215

### How to test

- Log in as a state user
- Navigate to the MLR report and create one
- Fill in all the fields (including the Page 2 modal)
- Submit the report
- Log in as an admin user
- Navigate to the MLR report
- Archive and/or release the report

**Note:** When you save the form in the Page 2 modal, it won't be shown on the page but it will be saved on the S3 bucket and DynamoDB table. Functionality for presenting these forms on the UI is related to a separate ticket. 

### Important updates

Never fear, MLR is here!

### Author checklist

- [ ] I have performed a self-review of my code
- [ ] I have added [thorough](https://bit.ly/3zPrxuZ) tests, if necessary
- [ ] I have updated the documentation, if necessary

### CODE REVIEW

**Comment by ntsummers1** [comment]
Entering a program doesn't show its name
Screenshot 2023-03-17 at 11 21 35 AM

On submission screen the name is undefined as well, and need to change mcpar to mlr
Screenshot 2023-03-17 at 11 24 48 AM

Due date is NaNaNa, maybe something more descriptive?

Screenshot 2023-03-17 at 11 18 10 AM


Probably dont need the exclusion of MCPAR note here as well
Screenshot 2023-03-17 at 11 20 14 AM

Can we update this error message for when a report has been submitted that we can't edit the name? Or to remove that ability if its been submitted?

https://user-images.githubusercontent.com/19439679/225947943-f781fc62-620d-40d4-9ccd-3e95beb8a7a7.mov

Admins can update the resubmission field but not actually save it. Please tell me this isn't autosave failing you 😓 

https://user-images.githubusercontent.com/19439679/225948871-0d51e172-1f0e-4c70-b9a8-422575cfd11f.mov



Double Checking: Should this page show my answers and more yet? Or to be done at a later time?
Screenshot 2023-03-17 at 11 18 56 AM


Should the print screen show the total reports submitted yet? Or to be done at a later time?

Screenshot 2023-03-17 at 11 17 08 AM

**Review by ntsummers1 — CHANGES_REQUESTED** [review]
See notes above - Feel free to tell me i've overstepped on any of these. Apologies if I have. Overall this is some really awesome work!

**Comment by braxex** [comment]
@ntsummers1 Great notes above. Will you split those out into individual comments so they can be addressed and closed individually?

**Comment by ntsummers1** [comment]
Screenshot 2023-03-22 at 12 03 10 PM

Should this still have the date and MCPAR section when a user goes to the print view?

**Comment by ntsummers1** [comment]
Screenshot 2023-03-22 at 12 04 07 PM

I believe this should say 1 MLR reports total underneath here, could be wrong

**Comment by oweingrod-coforma** [comment]
> Screenshot 2023-03-22 at 12 04 07 PM
> 
> I believe this should say 1 MLR reports total underneath here, could be wrong

Currently it shows 0 before you add a program report and then correctly displays the number of reports (but not the reports themselves, that's another ticket).

**Comment by ntsummers1** [comment]
Screenshot 2023-03-22 at 12 07 18 PM

These buttons have to be a color contrast issue for accessibility right? Or is it because they are disabled its ok? I suspect we'd have to change the disabled state here to match what the new submission button looks like for mcpar status page or some other variation that design might recommend

**Comment by oweingrod-coforma** [comment]
> Screenshot 2023-03-22 at 12 07 18 PM
> 
> These buttons have to be a color contrast issue for accessibility right? Or is it because they are disabled its ok? I suspect we'd have to change the disabled state here to match what the new submission button looks like for mcpar status page or some other variation that design might recommend

That's a good shout, let me check with design

File: services/ui-src/src/components/pages/Dashboard/DashboardPage.test.tsx [hunk 1]
```diff
@@ -102,34 +110,38 @@ describe("Test Dashboard view (with reports, desktop view)", () => {
...
     expect(mockUseNavigate).toBeCalledTimes(1);
     expect(mockUseNavigate).toBeCalledWith("/mock/mock-route-1");
   });
 
-  test("Clicking 'Add a Program' button opens the AddEditProgramModal", async () => {
-    const addProgramButton = screen.getByText(verbiage.body.callToAction);
-    expect(addProgramButton).toBeVisible();
-    await userEvent.click(addProgramButton);
-    await expect(screen.getByTestId("add-edit-program-form")).toBeVisible();
+  // TODO: test MLR
```
Comment (ntsummers1) [inline services/ui-src/src/components/pages/Dashboard/DashboardPage.test.tsx:131]: Can likely remove this

**Review by ntsummers1 — DISMISSED** [review]

**Pull request merged**